packages/devtools_app/lib/src/ui/search.dart

```diff
@@ -380,55 +380,20 @@ class AutoCompleteState extends State<AutoComplete> with AutoDisposeMixin {
     TextStyle regularTextStyle,
     TextStyle highlightedTextStyle,
   ) {
-    final text = match.text;
-    final matchedSegments = match.matchedSegments;
-
-    if (matchedSegments == null || matchedSegments.isEmpty) {
-      return TextSpan(
-        text: text,
+    return match.transformAutoCompleteMatch<TextSpan>(
+      transformMatchedSegment: (segment) => TextSpan(
+        text: segment,
+        style: highlightedTextStyle,
+      ),
+      transformUnmatchedSegment: (segment) => TextSpan(
+        text: segment,
         style: regularTextStyle,
-      );
-    }
-
-    final spans = <TextSpan>[];
-    int previousEndIndex = 0;
-
-    for (final segment in matchedSegments) {
-      if (previousEndIndex < segment.begin) {
-        // Add the unhighlighted segment before the current highlighted segment:
-        final segmentBefore = text.substring(previousEndIndex, segment.begin);
-        spans.add(
-          TextSpan(
-            text: segmentBefore,
-            style: regularTextStyle,
-          ),
-        );
-      }
-      // Add the current highlighted segment:
-      final highlightedSegment = text.substring(segment.begin, segment.end);
-      spans.add(
-        TextSpan(
-          text: highlightedSegment,
-          style: highlightedTextStyle,
-        ),
-      );
-      previousEndIndex = segment.end;
-    }
-    if (previousEndIndex < text.length - 1) {
-      // Add the last unhighlighted segment:
-      final lastSegment = text.substring(previousEndIndex);
-      spans.add(
-        TextSpan(
-          text: lastSegment,
-          style: regularTextStyle,
-        ),
-      );
-    }
-
-    return TextSpan(
-      text: spans.first.text,
-      style: spans.first.style,
-      children: spans.sublist(1),
+      ),
+      combineSegments: (segments) => TextSpan(
+        text: segments.first.text,
```

Comment:
We should have some length checks on `segments`. `.first` and `.sublist` will throw exceptions if the list is empty

Reply:
Good point, added an `assert` in the `transformAutocompleteMatch` method to verify segments is not empty.